stan/math/prim/prob/ddm_lpdf.hpp

```diff
@@ -0,0 +1,382 @@
+  } else {
+    for (size_t i = 0; i < Na; i++) {
+      if (a_vec[i] > 0) {
+        if (isfinite(a_vec[i])) {
+          continue;
+        } else {  // a = Inf implies PDF = log(0)
+          throw_domain_error(function, "a (threshold separation)", a_vec[i],
+                             " = ", ", but it must be positive and finite");
+        }
+      } else {  // {NaN, NA} evaluate to FALSE
+        throw_domain_error(function, "a (threshold separation)", a_vec[i],
+                           " = ", ", but it must be positive and finite");
+      }
+    }
+  }
+
+  if (Nv < 1) {  // v
+    return 0;
+  } else {
+    for (size_t i = 0; i < Nv; i++) {
+      if (isfinite(v_vec[i])) {
+        continue;
+      } else {  // NaN, NA, Inf, -Inf are not finite
+        throw_domain_error(function, "v (drift rate)", v_vec[i], " = ",
+                           ", but it must be finite");
+      }
+    }
+  }
+
+  if (Nt0 < 1) {  // t0
+    return 0;
+  } else {
+    for (size_t i = 0; i < Nt0; i++) {
+      if (t0_vec[i] >= 0) {
+        if (isfinite(
+                t0_vec[i])) {  // this could also be handled in calculate_pdf()
+          continue;
+        } else {  // t0 = Inf implies rt - t0 < 0 implies PDF = log(0)
+          throw_domain_error(function, "t0 (non-decision time)", t0_vec[i],
+                             " = ", ", but it must be positive and finite");
+        }
+      } else {  // {NaN, NA} evaluate to FALSE
+        throw_domain_error(function, "t0 (non-decision time)", t0_vec[i], " = ",
+                           ", but it must be positive and finite");
+      }
+    }
+  }
+
+  if (Nw < 1) {  // w
+    return 0;
+  } else {
+    for (size_t i = 0; i < Nw; i++) {
+      if (w_vec[i] > 0 && w_vec[i] < 1) {
+        continue;
+      } else {  // {NaN, NA} evaluate to FALSE
+        throw_domain_error(function, "w (relative a priori bias)", w_vec[i],
+                           " = ", ", but it must be that 0 < w < 1");
+      }
+    }
+  }
+
+  if (Nsv < 1) {  // sv
+    return 0;
+  } else {
+    for (size_t i = 0; i < Nsv; i++) {
+      if (sv_vec[i] >= 0) {
+        if (isfinite(sv_vec[i])) {
+          continue;
+        } else {  // sv = Inf implies PDF = log(0)
+          throw_domain_error(
+              function, "sv (standard deviation of drift rate across trials)",
+              sv_vec[i], " = ", ", but it must be positive and finite");
+        }
+      } else {  // {NaN, NA} evaluate to FALSE
+        throw_domain_error(
+            function, "sv (standard deviation of drift rate across trials)",
+            sv_vec[i], " = ", ", but it must be positive and finite");
+      }
+    }
+  }
+
+  if (!include_summand<propto, T_rt, T_response, T_a, T_v, T_t0, T_w,
+                       T_sv>::value) {
+    return 0;
+  }
+
+  // Calculate log(PDF)
+  T_return_type lp(0.0);
+  double t, a_i, v_i, w_i, sv_i;
+  for (size_t i = 0; i < Nmax; i++) {
+    // Check Parameter Values
+    t = rt_vec[i % Nrt]
```

Comment:
The pattern of indexing into `rt_vec` isn't right.

The purpose of the view (`scalar_seq_view<T_rt_ref> rt_vec(rt_ref)`) is to treat `rt_vec` as if it were a vector even if it's a scalar.

So... this can be simplified to something like:
```
t = rt_vec[i] - t0_vec[i];
```

But... when you see that, you'll see the types are wrong now. `t` isn't a `double`, right? It could either be `double` or `var` (or `fvar<T>`). I don't think this should work if you've tested this with `rt` as `var` and `t0` as `double`.